src/Dotnet.Installer.Core/Services/Implementations/SnapService.cs

```diff
@@ -10,23 +10,21 @@ public bool IsSnapInstalled(string name, CancellationToken cancellationToken = d
         return Directory.Exists(Path.Combine("/", "snap", name));
     }
 
-    public async Task<InvocationResult> Install(string name, CancellationToken cancellationToken = default)
+    public Task<Terminal.InvocationResult> Install(string name, CancellationToken cancellationToken = default)
     {
-        var result = await Terminal.Invoke("snap", "install", name);
-        return new InvocationResult(result == 0, "", "");
+        return Terminal.Invoke("snap", "install", name);
     }
 
-    public async Task<InvocationResult> Remove(string name, bool purge = false, CancellationToken cancellationToken = default)
+    public Task<Terminal.InvocationResult> Remove(string name, bool purge = false, CancellationToken cancellationToken = default)
     {
         var arguments = new List<string>
         {
             "remove"
         };
-        
+
         if (purge) arguments.Add("--purge");
         arguments.Add(name);
-        
-        var result = await Terminal.Invoke("snap", arguments.ToArray());
-        return new InvocationResult(result == 0, "", "");
+
+        return Terminal.Invoke("snap", arguments.ToArray());
```

Comment:
note: This was confusing to read, because I am used to the Async suffix of methods that return a Task or are marked `async` 😅

Reply:
Most of the methods in this project are either `async` or return `Task<T>`. I decided not to suffix them with `Async` because they would *mostly* be `*Async` 😆 

We can discuss this further and refactor the method names if necessary.